server/log_operation_manager.go

```diff
@@ -90,124 +90,30 @@ type LogOperationInfo struct {
 	// batch takes longer than this interval to complete, the next batch
 	// will start immediately.
 	RunInterval time.Duration
-	// PreElectionPause is the maximum interval to wait before starting a
-	// mastership election for a particular log.
-	PreElectionPause time.Duration
-	// MasterCheckInterval is the interval between checks that we still
-	// hold mastership for a log.
-	MasterCheckInterval time.Duration
+
+	// ElectionConfig configures per-log master election process.
+	ElectionConfig election.Config
 	// MasterHoldInterval is the minimum interval to hold mastership for.
 	MasterHoldInterval time.Duration
-	// ResignOdds gives the chance of resigning mastership after each
-	// check interval, as the N for 1-in-N.
-	ResignOdds int
+	// ResignSpread is the max extra number of master check intervals to keep
+	// mastership for. The master will resign in:
+	//   MasterHoldInterval + rand(ResignSpread) * MasterCheckInverval.
+	ResignSpread int
+
 	// NumWorkers is the number of worker goroutines to run in parallel.
 	NumWorkers int
 }
 
-type electionRunner struct {
-	logID    int64
-	info     *LogOperationInfo
-	tracker  *election.MasterTracker
-	cancel   context.CancelFunc
-	wg       *sync.WaitGroup
-	election election.MasterElection
-}
-
-type resignation struct {
-	er   *electionRunner
-	done chan<- bool
-}
-
-func (r *resignation) execute(ctx context.Context) {
-	glog.Infof("%d: deliberately resigning mastership", r.er.logID)
-	if err := r.er.election.Resign(ctx); err != nil {
-		glog.Errorf("%d: failed to resign mastership: %v", r.er.logID, err)
+// getResignDelay computes master resignation delay based on operation
+// parameters.
+func getResignDelay(info *LogOperationInfo) time.Duration {
```

Comment:
Can you rename this to something not starting with 'get', just `resignDelay` for example.